### ISSUE TITLE
Cache active check is broken

### ISSUE THREAD

**Issue by scottsb** [wigman/AjaxSwatches#32]
I'm not entirely sure the intention of this method that's in both the `List` and `Swatchlist` blocks, but whatever it does, it has no effect:

``` php
protected function _isCacheActive()
{

    /* if there are any messages dont read from cache to show them */
    if (Mage::getSingleton('core/session')->getMessages(true)->count() > 0) {
        return false;
    }
    return true;

}
```

Because where it's called in both classes (inside `getCacheKey`) it triggers the parent class's `getCacheKey` method, but its return value is simply discarded, allowing control to pass through to the rest of the method, regardless of what had been returned from `_isCacheActive`:

``` php
if (!$this->_isCacheActive()) {
    parent::getCacheKey();
}
```

**Comment by wigman** [comment]
Closing this one since it's a duplicate of https://github.com/wigman/AjaxSwatches/issues/35.

**Comment by scottsb** [comment]
@wigman, I don't think this actually is a duplicate. This is a bug that occurs in both the LIst and Swatchlist blocks, but it doesn't have anything do with the relationship between the blocks.

**Comment by wigman** [comment]
@scottsb You're right, I only looked at the titles of the tickets. Apologies!

However, this piece of code does seem to work for me:) As a matter of fact, I've copied the principle from this module: https://github.com/netresearch/Magento-CatalogCache/blob/master/app/code/community/Netresearch/CatalogCache/Block/Layer/View.php

It falls back to the parent method in case $this->_isCacheActive() returns false. Otherwise it continues with the custom CacheTags.
